sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/AzureServiceClient.java

```diff
@@ -114,10 +114,8 @@ public Context getContext() {
      * @return the merged context.
      */
     public Context mergeContext(Context context) {
-        for (Map.Entry<Object, Object> entry : this.getContext().getValues().entrySet()) {
-            context = context.addData(entry.getKey(), entry.getValue());
-        }
-        return context;
+        // data from context would override data from client
+        return CoreUtils.mergeContexts(this.getContext(), context);
```

Comment:
Context is immutable.

Reply:
Is this the only place that merges context?

Reply:
That is all, AFAIK.

Lite would be changed in codegen https://github.com/Azure/autorest.java/pull/1445